element-hq/element-x-android
# Add full screen intent permissions banner

## Type of change


## Content

- Add `FullScreenIntentPermissionsPresenter` to handle the logic of checking the permission state and handling actions related to it.
- Add a separate set of modules for it.
- Use the right package name for the contents of `:libraries:permissions` (it was using `io.element.android.features.*` before).

## Motivation and context

For Android 14+ we may need to manually ask permissions for full screen intents.

## Screenshots / GIFs

|Before|After|
|-|-|
|![](https://media.githubusercontent.com/media/element-hq/element-x-android/6b243228de93822b4765a83bc17a2b91340c6cea/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.preferences.impl.notifications_NotificationSettingsView_null_NotificationSettingsView-Day-6_7_null_11%2CNEXUS_5%2C1.0%2Cen%5D.png)|![](https://media.githubusercontent.com/media/element-hq/element-x-android/4084d2d8756d36f834317d79ceed5bfc77331511/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.preferences.impl.notifications_NotificationSettingsView_null_NotificationSettingsView-Day-6_7_null_11%2CNEXUS_5%2C1.0%2Cen%5D.png)|
||![](https://media.githubusercontent.com/media/element-hq/element-x-android/4084d2d8756d36f834317d79ceed5bfc77331511/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.roomlist.impl.components_FullScreenIntentPermissionBanner_null_FullScreenIntentPermissionBanner-Day-3_4_null%2CNEXUS_5%2C1.0%2Cen%5D.png)|

## Tests



- On an Android 14+ device, open the app. A banner should be displayed in the room list.
- Go to settings -> notifications. Check a similar banner is found there.
- Open the settings by tapping on the banner, enable permissions.
- Check the banner is now gone.
- Using the recents menu go back to the permissions screen opened before, disable the permission and return to the app.
- Check the banner is back.
- Go back to the room list, close the banner.
- If you restart the app it should still be gone, but if you go to settings > notification the warning should still be there.

## Tested devices

- OS version(s): 14, 15.

## Checklist
